Repo jbei org changes

jdmccauley opened the pull request:
* Remove self-import

The predict_loading_concentration module was importing itself, which is
probably not what we want it doing.

* Wrap top-level predict_loading_concentration in main

jdmccauley committed:
Initial commit to get working

wmorrell committed:
Remove self-import

The predict_loading_concentration module was importing itself, which is
probably not what we want it doing.

wmorrell committed:
Wrap top-level predict_loading_concentration in main

jdmccauley committed:
Revert README and delete files

Remove the run.sh and Pipfile for this repo since they are not used. Also, revert the README to its previous state in commit 651b2c381357deb.

tianar commented on predict_loading_concentration.py:
```diff
@@ -4,35 +4,39 @@ def warn(*args, **kwargs):
 import warnings
```
This file should be deleted altogether. It is from some previous version (you see a different package name `diva_dna_seq`). The one that is used is diva_seq_opt/predict_loading_concentration.py.

jdmccauley replied:
This was much more than that...
The predict_loading_concentration() was in the __init__.py file, and the main predict_loading_concentration.py seems more recent at the parent directory level. See most recent commit.

There is still room for improvement, but at least now we have only one predict_loading_concentration.py, and the predict_loading_concentration() is in the utility.py, not in __init__.py.

tianar replied:
Good progress. But now there will be a problem in setup.py file, which used to work with `predict_loading_concentration.py` file that was in `diva_seq_opt` dir. See line 11 in setup.py. 
What I would do is change to
``
entry_points = {        
'console_scripts': ['predict-loading-concentration=predict_loading_concentration:main'],      
},
``
This should be tested after a fresh install. 
Note the change `_`->`-` in the command line tool. I would do that to distinguish from the module and function with the same name, but I leave it to you.

jdmccauley replied:
For now I think keeping the underscores might be better for the sole purpose of more shell support. I just read that some shells don't allow dashes in function names:
https://unix.stackexchange.com/questions/482417/will-it-be-bad-that-a-function-or-script-name-contains-dash-instead-of-under

tianar reviewed (changes requested):
Good start!

jdmccauley committed:
Found the real predict_loading_concentration

The real predict_loading_concentration function that was called in the two predict_loading_concentration.py files was located in __init__.py.
This real function was moved to diva_seq_opt/utility.
The duplicated predict_loading_concentration.py in diva_seq_opt was deleted, since the same actions are done by the parent directory's predict_loading_concentration.py.
The __init__.py was cleared of all text, so it serves only to make diva_seq_opt a module.
The main predict_loading_concentration.py was modified to make the call to predict_loading_concentration() in utility.py.

tianar commented on predict_loading_concentration.py:
```diff
@@ -4,35 +4,37 @@ def warn(*args, **kwargs):
 
+def main():
+	#Allow Matplotlib to be used on commandline
+	import matplotlib as mpl
```
I think this should be outside of main()

jdmccauley replied:
Great catch!

tianar commented on predict_loading_concentration.py:
```diff
@@ -4,35 +4,37 @@ def warn(*args, **kwargs):
-parser.add_argument('OF' ,type=str, help='Prediction Plot output file, *.png')
-args = parser.parse_args()
+	#Load Model
+	with open('./diva_seq_opt/model/model30.pkl','rb') as fp:
```
Later on we should change this model name to e.g. model_latest.pkl and always have a copy of the latest model. For example, at the end of the year the `model` directory will contain:
model_2018.pkl, model_05_2021.pkl, model_12_2021.pkl, model_latest.pkl, where the last two would be the same.

jdmccauley replied:
I very much agree for the future!

tianar replied:
Can you please include a TODO about that before this line?

tianar replied:
I think the path for models should not be among the code but in a separate directory at the same level as notebooks.

tianar reviewed (changes requested):
See my comments inline

tianar commented:
Also, please check the README, in particular the installation instructions.

jdmccauley committed:
Move import, update setup/README

Moved an import statement in predict_loading_concentration:main from main() to the top.
Edited setup.py to call the new predict_loading_concentration.
Iterated version.
Updated REAMDE to show dependency installation instructions.
Finish the last sentence of the README.

tianar commented on README.md:
````diff
@@ -4,12 +4,11 @@ This package is a repository of tools used by the DIVA DNA Seq team at the Joint
-pip install diva_dna_seq
+The dependencies of this tool can be installed with:
+```
+pip install -r requirements.txt
````
What about installing the package using pip? https://pypi.org/project/diva-seq-opt/

That done using setup.py. I can help with that if you'd like.

tianar commented on install.sh:
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+# Installs the pipenv for running the tool.
```
There should be instructions for this in README.

tianar reviewed (changes requested):
Getting there!
